ygen/enumgen.go

```diff
@@ -176,7 +181,11 @@ func (s *enumSet) enumeratedTypedefTypeName(args resolveTypeArgs, prefix string,
 	// types which is defined in RFC6020/RFC7950) then we establish what the type
 	// that we must actually perform the mapping for is. By default, start with
 	// the type that is specified in the schema.
-	if !util.IsYANGBaseType(args.yangType) {
+	// If the type is a built-in enumeration residing in a typedef union
+	// (indicated by args.subsumingType), and useDefiningModuleForTypedef
+	// is set, then it's also named as a typedef, with its name based on
+	// the union typedef's name.
+	if !util.IsYANGBaseType(args.yangType) || (useDefiningModuleForTypedefEnumNames && args.yangType.Kind == yang.Yenum && args.subsumingType != nil && !util.IsYANGBaseType(args.subsumingType)) {
```

Comment:
I'd suggest a helper function (could be in-lined) for this:

```
(useDefiningModuleForTypedefEnumNames && args.yangType.Kind == yang.Yenum && args.subsumingType != nil && !util.IsYANGBaseType(args.subsumingType))
```

Can you explain in what circumstances `util.IsYANGBaseType(args.subsumingType)` can ever return true for a `args.subsumingType`? I assume that it's the case where we have a union that is not a typedef union, right - but I'm not sure we would ever exercise this code path for those.

Reply:
PTAL with `DefiningType`.